view/js/dropzone-factory.js

```diff
@@ -0,0 +1,63 @@
+var DzFactory = function () {
+	this.createDropzone = function(element, target, maxImagesize) {
+		return new Dropzone( element, {
+			paramName: 'userfile', // The name that will be used to transfer the file
+			maxFilesize: maxImagesize, // MB
+			url: '/media/photo/upload?response=url&album=',
+			accept: function(file, done) {
+				done();
+			},
+			init: function() {
+				this.on('success', function(file, serverResponse) {
+					var _target = $(target)
+					var resp = $(serverResponse).find('div#content').text()
+					if (_target.setRangeText) {
+						//if setRangeText function is supported by current browser
+						_target.setRangeText(' ' + $.trim(resp) + ' ')
+					} else {
+						_target.focus()
+						document.execCommand('insertText', false /*no UI*/, ' ' + $.trim(resp) + ' ');
+					}
```

Comment:
A few things here:
- Please avoid the overly-general `var` in favor of `let` (same meaning as PHP) or `const` (if the value isn't re-initialized.
- Please add a semi-colon (;) at the end of each statement even if it isn't necessary.
- Please avoid vague variable names like "target" or "response" in favor of more specific names.
- You can combine the string operations of `bbcodeString` in one place.
- I like to prepend variable names with a dollar sign when it contains a jQuery object. This also allowed me to realize you are trying to call [`HTMLInputElement.setRangeText`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/setRangeText) on a jQuery object, which probably won't work. Please review and address.
```suggestion
					const $targetTextarea = $(textareaSelector);
					const bbcodeString = ' ' + $.trim($(serverResponse).find('div#content').text()) + ' ';
					if ($targetTextarea.setRangeText) {
						// if setRangeText function is supported by current browser
						$targetTextarea.setRangeText(bbcodeString);
					} else {
						$targetTextarea.focus();
						document.execCommand('insertText', false /*no UI*/, bbcodeString);
					}
```